src/txn/txn_rollback_to_stable.c

```diff
@@ -629,16 +774,44 @@ __rollback_abort_row_ondisk_kv(
  */
 static void
 __rollback_abort_newer_col_var(
-  WT_SESSION_IMPL *session, WT_PAGE *page, wt_timestamp_t rollback_timestamp)
+  WT_SESSION_IMPL *session, WT_REF *ref, wt_timestamp_t rollback_timestamp)
 {
+    WT_CELL *kcell;
+    WT_CELL_UNPACK_KV unpack;
     WT_COL *cip;
     WT_INSERT_HEAD *ins;
+    WT_PAGE *page;
+    uint64_t recno, rle;
     uint32_t i;
+    bool stable_update_found;
+
+    page = ref->page;
+    /*
+     * If a disk image exists, start from the provided recno; or else start from 0.
+     */
+    if (page->dsk != NULL)
+        recno = page->dsk->recno;
+    else
+        recno = 0;
+
+    /* Review the changes to the original on-page data items. */
+    WT_COL_FOREACH (page, cip, i) {
+        stable_update_found = false;
 
-    /* Review the changes to the original on-page data items */
-    WT_COL_FOREACH (page, cip, i)
         if ((ins = WT_COL_UPDATE(page, cip)) != NULL)
-            __rollback_abort_newer_insert(session, ins, rollback_timestamp);
+            __rollback_abort_col_newer_insert_update(
+              session, ins, rollback_timestamp, &stable_update_found);
+
+        if (!stable_update_found && page->dsk != NULL) {
+            kcell = WT_COL_PTR(page, cip);
+            __wt_cell_unpack_kv(session, page->dsk, kcell, &unpack);
+            rle = __wt_cell_rle(&unpack);
+            __rollback_abort_col_ondisk_kv(session, ref, cip, rollback_timestamp, recno);
```

Comment:
Should col var also check the history store and restore the update from the history store?

Reply:
That's in the scope of another ticket which will be completed once this ticket is finished.